subsys/net/lib/dhcpv6/dhcpv6.c

```diff
@@ -535,7 +539,7 @@ static int dhcpv6_add_options(struct net_if *iface, struct net_pkt *pkt,
 		if (options->iaprefix) {
 			memcpy(&ia_pd.iaprefix.prefix, &iface->config.dhcpv6.prefix,
 			       sizeof(ia_pd.iaprefix.prefix));
-			ia_pd.iaprefix.prefix_len = iface->config.dhcpv6.prefix_len;
+			ia_pd.iaprefix.prefix_len = CONFIG_NET_CONFIG_DHCPV6_REQUESTED_PREFIX_LEN;
```

Comment:
DHCP module should have no dependencies to `NET_CONFIG` module, the latter it totally optional. It seems that net config should set `iface->config.dhcpv6.prefix_len` option instead?